spyder_line_profiler/spyder/widgets.py

```diff
@@ -19,41 +16,34 @@
 import os
 import os.path as osp
 import time
-import sys
+from datetime import datetime
 
 # Third party imports
-from qtpy.compat import getopenfilename
-from qtpy.QtCore import (QByteArray, QProcess, Qt, QTextCodec,
-                         QProcessEnvironment, Signal)
+from qtpy.QtWidgets import QHBoxLayout, QLabel
 from qtpy.QtGui import QBrush, QColor, QFont
-from qtpy.QtWidgets import (QHBoxLayout, QWidget, QMessageBox, QVBoxLayout,
-                            QLabel, QTreeWidget, QTreeWidgetItem, QApplication)
+from qtpy.QtCore import (QByteArray, QProcess, Qt, QTextCodec,
+                         QProcessEnvironment, Signal, QTimer)
+from qtpy.QtWidgets import (QMessageBox, QVBoxLayout,
+                            QTreeWidget, QTreeWidgetItem, QApplication)
+from qtpy.compat import getopenfilename, getsavefilename
+import qtawesome as qta
+
+# Spyder imports
 
-# Local imports
-from spyder.config.base import get_conf_path, get_translation
+from spyder.api.config.decorators import on_conf_change
+from spyder.api.translations import get_translation
+from spyder.config.base import get_conf_path
 from spyder.plugins.variableexplorer.widgets.texteditor import TextEditor
-from spyder.utils import programs
-from spyder.utils.misc import add_pathlist_to_PYTHONPATH
-from spyder.utils.qthelpers import create_toolbutton, get_icon
+from spyder.api.widgets.main_widget import PluginMainWidget
 from spyder.widgets.comboboxes import PythonModulesComboBox
+from spyder.utils import programs
+from spyder.utils.misc import add_pathlist_to_PYTHONPATH, getcwd_or_home
+from spyder.plugins.run.widgets import get_run_configuration
+from spyder.py3compat import to_text_string, getcwd, pickle
+from spyder.utils.icon_manager import ima
```

Comment:
Seems like `getcwd` and `ima` imports are not being used:
```suggestion
from spyder.py3compat import to_text_string, pickle
```